src/front/glsl/lex_pp.rs

```diff
@@ -0,0 +1,242 @@
+use super::{parser::Token, token::TokenMetadata, types::parse_type};
+use crate::FastHashMap;
+use pp_rs::{
+    pp::Preprocessor,
+    token::{Punct, Token as PPToken, TokenValue},
+};
+use std::collections::VecDeque;
+
+//#[derive(Clone, Debug)]
+pub struct LexerPP<'a> {
```

Comment:
similarly, if we don't have the regular `Lexer` any more, we can just call this struct `Lexer` instead of `LexerPP`.